src/modules/AuthModule/components/Register/Register.tsx

```diff
@@ -1,9 +1,324 @@
-import React from 'react'
+import { useState } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { Form, useNavigate, Link } from "react-router-dom";
+import { toast } from "react-toastify";
+import imgRegister from "../../../../assets/images/Register.png";
+import styleRegister from "./Register.module.css";
+import { FormDataRegister } from "../../../../interfaces/Auth";
+import { axiosInstance } from "../../../../axiosConfig/axiosInstance";
+import Visibility from "@mui/icons-material/Visibility";
+import VisibilityOff from "@mui/icons-material/VisibilityOff";
+import {
+  Alert,
+  Box,
+  Button,
+  CircularProgress,
+  Container,
+  FilledInput,
+  FormControl,
+  Grid,
+  IconButton,
+  InputAdornment,
+  TextField,
+  Typography,
+} from "@mui/material";
+import DriveFolderUploadIcon from "@mui/icons-material/DriveFolderUpload";
+import {
+  emailValidation,
+  passwordValidation,
+} from "./../../../../validations/validations";
+export default function Register() {
+  const navigate = useNavigate();
+  const [loadingBtn, setLoadingBtn] = useState(false);
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+
+  // upload img
+  const [fileInputContent, setFileInputContent] = useState(
+    "Drag & Drop or Choose a Item Image to Upload"
+  );
+  const handleInputContent = () => {
+    setFileInputContent("File Uploaded Successfully");
+  };
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<FormDataRegister>();
```

Comment:
This is actually not a good practice to have multiple packages that perform the same functionality. Here we are using react-hook-form but in other components we use Formik. Managing dependencies and avoiding any unneeded ones is some crucial to avoid having an application that is too large because of having too many unneeded packages.